### PR TITLE
Find *or* create auction ended job on auction update

### PR DESCRIPTION
**Pull request by jessieay**
* If there is not aready a job, create one!
* Prevents issue with existing records in DB

### CODE REVIEW

File: app/services/update_auction.rb [hunk 1]
```diff
@@ -29,21 +29,26 @@ def assign_attributes
 
   def update_auction_ended_job
     if updating_ended_at?
-      job = auction_ended_job
-
-      if job
-        job.update(run_at: auction.ended_at)
+      if find_auction_ended_job
+        find_auction_ended_jobjob.update(run_at: auction.ended_at)
```
Comment (adelevie) [inline app/services/update_auction.rb]: typo?

Comment (jessieay) [inline app/services/update_auction.rb]: yup! fixed :)

File: spec/services/update_auction_spec.rb [hunk 2]
```diff
@@ -3,31 +3,54 @@
...
-        params = { auction: new_ended_at}
-
-        expect(job.run_at).to eq(auction.ended_at)
-        UpdateAuction.new(auction: auction,
-                          params: params,
-                          current_user: auction.user).perform
-
-        job.reload
-        expect(job.run_at).to eq(parsed_new_ended_at)
+      context 'job not enqueue' do
```
Comment (adelevie) [inline spec/services/update_auction_spec.rb]: should be `job not enqueued`?

File: app/services/update_auction.rb [hunk 3]
```diff
@@ -29,21 +29,26 @@ def assign_attributes
 
   def update_auction_ended_job
     if updating_ended_at?
-      job = auction_ended_job
-
-      if job
-        job.update(run_at: auction.ended_at)
+      if find_auction_ended_job
+        find_auction_ended_job.update(run_at: auction.ended_at)
```
Comment (adelevie) [inline app/services/update_auction.rb]: yaaay

File: spec/services/update_auction_spec.rb (hunk 2 above)
Comment (jessieay) [inline spec/services/update_auction_spec.rb]: good call - fixed!

File: spec/services/update_auction_spec.rb [hunk 4]
```diff
@@ -3,31 +3,54 @@
...
-        params = { auction: new_ended_at}
-
-        expect(job.run_at).to eq(auction.ended_at)
-        UpdateAuction.new(auction: auction,
-                          params: params,
-                          current_user: auction.user).perform
-
-        job.reload
-        expect(job.run_at).to eq(parsed_new_ended_at)
+      context 'job not enqueued' do
```
Comment (adelevie) [inline spec/services/update_auction_spec.rb]: w00t

File: spec/services/update_auction_spec.rb (hunk 2 above)
Comment (jessieay) [inline spec/services/update_auction_spec.rb]: is there a word harder to type than "enqueue" ?

**Comment by adelevie** [comment]
looks GREAT

**Pull request merged**